java/runtime/src/main/java/io/ray/runtime/gcs/GcsClient.java

```diff
@@ -96,19 +96,16 @@ public GcsClient(String redisAddress, String redisPassword) {
   }
 
   private Map<String, Double> getResourcesForClient(UniqueId clientId) {
-    final String prefix = TablePrefix.NODE_RESOURCE.toString();
-    final byte[] key = ArrayUtils.addAll(prefix.getBytes(), clientId.getBytes());
-    Map<byte[], byte[]> results = primary.hgetAll(key);
-    Map<String, Double> resources = new HashMap<>();
-    for (Map.Entry<byte[], byte[]> entry : results.entrySet()) {
-      String resourceName = new String(entry.getKey());
-      Gcs.ResourceTableData resourceTableData;
-      try {
-        resourceTableData = Gcs.ResourceTableData.parseFrom(entry.getValue());
-      } catch (InvalidProtocolBufferException e) {
-        throw new RuntimeException("Received invalid protobuf data from GCS.");
-      }
-      resources.put(resourceName, resourceTableData.getResourceCapacity());
+    byte[] resourceMapBytes = globalStateAccessor.getNodeResourceInfo(clientId);
+    HashMap<String, Double> resources = new HashMap<>();
```

Comment:
move "HashMap<String, Double> resources = new HashMap<>();" to line 107 is better.

Reply:
> move "HashMap<String, Double> resources = new HashMap<>();" to line 107 is better.

resouces can not only in For-loop scope because it's return value of this function.